pallets/stableswap/src/lib.rs

```diff
@@ -1024,11 +1112,118 @@ pub mod pallet {
 				fee: Balance::zero(),
 			});
 
-			#[cfg(feature = "try-runtime")]
+			#[cfg(any(feature = "try-runtime", test))]
 			Self::ensure_remove_liquidity_invariant(pool_id, &initial_reserves);
 
 			Ok(())
 		}
+
+		/// Create a stable pool with a given list of assets and pegs.
+		///
+		/// This function allows the creation of a new stable pool with specified assets, amplification, fee, and peg sources. The pool is identified by a share asset.
+		///
+		/// Parameters:
+		/// - `origin`: Must be `T::AuthorityOrigin`.
+		/// - `share_asset`: Preregistered share asset identifier.
+		/// - `assets`: List of asset IDs to be included in the pool.
+		/// - `amplification`: Pool amplification parameter.
+		/// - `fee`: Fee to be applied on trade and liquidity operations.
+		/// - `peg_source`: Bounded vector specifying the source of the peg for each asset.
+		/// - `max_peg_update`: Maximum allowed peg update per block.
+		///
+		/// Emits `PoolCreated` event if successful.
+		/// Emits `AmplificationChanging` event if successful.
+		///
+		/// # Errors
+		/// - `IncorrectAssets`: If the assets are the same or less than 2 assets are provided.
+		/// - `MaxAssetsExceeded`: If the maximum number of assets is exceeded.
+		/// - `PoolExists`: If a pool with the given assets already exists.
+		/// - `ShareAssetInPoolAssets`: If the share asset is among the pool assets.
+		/// - `AssetNotRegistered`: If one or more assets are not registered in the AssetRegistry.
+		/// - `InvalidAmplification`: If the amplification parameter is invalid.
+		/// - `IncorrectInitialPegs`: If the initial pegs are incorrect.
+		/// - `MissingTargetPegOracle`: If the target peg oracle entry is missing.
+		/// - `IncorrectAssetDecimals`: If the assets have different decimals.
+		///
+		#[pallet::call_index(11)]
+		#[pallet::weight(<T as Config>::WeightInfo::create_pool_with_pegs())]
+		#[transactional]
+		pub fn create_pool_with_pegs(
+			origin: OriginFor<T>,
+			share_asset: T::AssetId,
+			assets: BoundedVec<T::AssetId, ConstU32<MAX_ASSETS_IN_POOL>>,
+			amplification: u16,
+			fee: Permill,
+			peg_source: BoundedPegSources,
```

Comment:
it should be noted that oracle source always loads price of first asset in poll vs target asset, mby this could be extended by pair specification as well. seems limiting otherwise